node/src/accountant/scanners.rs

```diff
@@ -2278,33 +2837,127 @@ mod tests {
 
-        subject.remove_timestamp(ScanType::Receivables, &logger);
+        subject.signal_scanner_completion(ScanType::Receivables, SystemTime::now(), &logger);
 
         TestLogHandler::new().exists_log_containing(&format!(
             "ERROR: {test_name}: Called scan_finished() for Receivables scanner but timestamp was not found"
         ));
     }
+
+    #[test]
+    fn signal_scanner_completion_does_not_allow_to_display_less_than_1_ms_of_scanner_duration() {
+        init_test_logging();
+        let test_name = "signal_scanner_completion_does_not_allow_to_display_less_than_1_ms_of_scanner_duration";
+        let logger = Logger::new(test_name);
+        let mut subject = ScannerCommon::new(Rc::new(make_custom_payment_thresholds()));
+        let now = SystemTime::now();
+        subject.initiated_at_opt = Some(now);
+
+        subject.signal_scanner_completion(ScanType::Payables, now, &logger);
+
+        TestLogHandler::new().exists_log_containing(&format!(
+            "INFO: {test_name}: The Payables scan ended in 1ms."
+        ));
+    }
+
+    fn assert_sound_elapsed_time_in_mark_as_ended<S: Message, T: Message>(
+        subject: &mut dyn Scanner<S, T>,
+        scanner_name: &str,
+        test_name: &str,
+        logger: &Logger,
+        log_handler: &TestLogHandler,
+        digits_capturing_regex: &Regex,
+    ) {
+        fn flip_0_to_1_or_keep(num: u128) -> u128 {
+            match num {
+                0 => 1,
+                x => x,
+            }
+        }
+
+        let before = SystemTime::now();
+        subject.mark_as_started(before);
+
+        subject.mark_as_ended(&logger);
+
+        let after = SystemTime::now();
+        let idx = log_handler.exists_log_containing(&format!(
+            "INFO: {}: The {} scan ended in ",
+            test_name, scanner_name
+        ));
+        let our_log_msg = log_handler.get_log_at(idx);
+        let captures = digits_capturing_regex.captures(&our_log_msg).unwrap();
+        let millis_str = captures.get(1).unwrap().as_str();
+        let millis = millis_str.parse::<u128>().unwrap();
+        let actual_millis_checked = flip_0_to_1_or_keep(millis);
+        let max_millis_elapsed = after.duration_since(before).unwrap().as_millis();
+        let max_millis_elapsed_checked = flip_0_to_1_or_keep(max_millis_elapsed);
+        assert!(
+            actual_millis_checked <= max_millis_elapsed_checked,
+            "We expected the time elapsed ({}) to be equal or shorter to {}",
+            actual_millis_checked,
+            max_millis_elapsed_checked
+        )
+    }
+
+    #[test]
+    fn mark_as_ended_uses_the_right_time_reference_for_each_scanner() {
+        init_test_logging();
+        let test_name = "mark_as_ended_uses_the_right_time_reference_for_each_scanner";
+        let logger = Logger::new(test_name);
+        let log_handler = TestLogHandler::new();
+        let digits_capturing_regex = Regex::new(r#"scan ended in (\d*)ms"#).unwrap();
```

Comment:
This regex is the only one ever used; it could theoretically be inside the `assert_sound_elapsed` asserter.